application/docker/Dockerfile

```
@@ -19,18 +20,40 @@ RUN apt-get update  \
     && rm -rf /var/lib/apt/lists/* \
     && apt-get clean
 
-# Install Rust compiler
+# Install Rust toolchain
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
 ENV PATH="/root/.cargo/bin:${PATH}"
 
-# Install dependencies
+##########
+# CPU Base
+##########
+FROM backend-base AS cpu-base
+
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=backend/uv.lock,target=uv.lock \
+    --mount=type=bind,source=backend/pyproject.toml,target=pyproject.toml \
+    uv sync --frozen --no-dev --no-editable --inexact --extra mqtt --extra cpu
```

```suggestion
    uv sync --frozen --no-dev --no-editable --extra mqtt --extra cpu
```
You can remove `--inexact`. I added it because we had two calls to `uv sync` (first one with `--extra mqtt` to install base deps, second one with `--inexact --extra cpu` to install CPU-specific deps): the `--inexact` flag in the second call ensured that sync would not uninstall the extra packages specified in the first call (mqtt).